lib/app/features/user/pages/profile_page/components/profile_details/followers_you_know/followed_by_text.dart

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/nostr/providers/nostr_cache.c.dart';
+import 'package:ion/app/features/user/model/follow_type.dart';
+import 'package:ion/app/features/user/model/user_metadata.c.dart';
+import 'package:ion/app/hooks/use_tap_gesture_recognizer.dart';
+import 'package:ion/app/router/app_routes.c.dart';
+
+class FollowedByText extends HookConsumerWidget {
+  const FollowedByText({
+    required this.pubkeys,
+    super.key,
+  });
+
+  final List<String> pubkeys;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final firstUserPubkey = pubkeys.first;
+
+    // User metadata is fetched alongside the `followersYouKnowDataSourceProvider`, so don't fetch it manually
+    final firstUserMetadata = ref.watch(
+      nostrCacheProvider.select(
+        cacheSelector<UserMetadataEntity>(
+          UserMetadataEntity.cacheKeyBuilder(pubkey: firstUserPubkey),
+        ),
+      ),
+    );
+
+    final userTapRecognizer = useTapGestureRecognizer(
+      onTap: () => ProfileRoute(pubkey: firstUserPubkey).push<void>(context),
+    );
+
+    final othersTapRecognizer = useTapGestureRecognizer(
+      onTap: () => FollowListRoute(pubkey: firstUserPubkey, followType: FollowType.followers)
+          .push<void>(context),
+    );
+
+    if (firstUserMetadata == null) {
+      return const SizedBox.shrink();
+    }
+
+    final defaultStyle = context.theme.appTextThemes.body2.copyWith(
+      color: context.theme.appColors.primaryText,
+    );
+
+    final actionableStyle = context.theme.appTextThemes.body2.copyWith(
+      color: context.theme.appColors.darkBlue,
+    );
+
+    final textSpans = [
```

Comment:
We can optimize widget rebuilds by memoizing properties that depend on other properties. This will help reduce unnecessary re-creation and improve performance. Does it makes sense for you?